drivers/gpio/Kconfig.it8801

```diff
@@ -0,0 +1,22 @@
+# Copyright (c) 2024 ITE Corporation. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+config GPIO_ITE_IT8801
+	bool "ITE IT8801 GPIO device driver"
+	default y
+	depends on DT_HAS_ITE_IT8801_GPIO_ENABLED
+	select I2C
+	select MFD
+	help
+	  Enable driver for ITE IT8801 I2C-based GPIO.
+
+if GPIO_ITE_IT8801
+
+config GPIO_IT8801_INIT_PRIORITY
+	int "IT8801 GPIO port init priority"
+	default 85
+	help
+	  IT8801 GPIO port device driver initialization priority. The priority
+	  must be lower than MFD_INIT_PRIORITY device.
```

Comment:
Add a BUILD_ASSERT to check the init priority.

Reply:
Use CONFIG_MFD_INIT_PRIORITY on all the child devices.